perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/FacilitiesManagerBlImpl.java

```diff
@@ -911,18 +917,67 @@ public void checkFacilityContactExists(PerunSession sess, Facility facility, Str
 	}
 
 	@Override
+	public List<Service> getAssignedServices(PerunSession sess, Facility facility) throws InternalErrorException {
+		return facilitiesManagerImpl.getAssignedServices(sess, facility);
+	}
+
+	@Override
 	public void assignSecurityTeam(PerunSession sess, Facility facility, SecurityTeam securityTeam) throws InternalErrorException {
 		facilitiesManagerImpl.assignSecurityTeam(sess, facility, securityTeam);
 		getPerunBl().getAuditer().log(sess, "{} was assigned to {}.", securityTeam, facility);
 	}
 
 	@Override
+	public void assignService(PerunSession sess, Facility facility, Service service) throws InternalErrorException, ServiceAlreadyAssignedException {
+		facilitiesManagerImpl.assignService(sess, facility, service);
+		getPerunBl().getAuditer().log(sess, "{} assigned to {}", service, facility);
+
+		AttributesManagerBl attributesManagerBl = getPerunBl().getAttributesManagerBl();
+		// call check of facility's resource's member's user's attributes
+		try {
+			attributesManagerBl.checkAttributesValue(sess, facility, attributesManagerBl.getRequiredAttributes(sess, facility));
```

Comment:
This seems to be correct from consistency point of view, but I'm afraid, that we could end up with service, which will never be assigned due to wrong value in attribute dependency with objects already connected with facility (user-member-resource). Some services shares attributes, also it's highly dependent on attribute modules check implementation. What do you think @licehammer ? @stavamichal suggested we use service connection on facility only to limit, what services can be configured on its resources. Usually until there is a resource and some groups assigned, there are no data to be pushed to service destination.